### PR TITLE
Optimizes UniqueName.update

### PR DESCRIPTION
**Pull request by kwiliarty**
Hi @amichal . Here's one of the optimizations we talked about. I believe that it's working, but I have a question about the way I'm assigning `all_names`. Rubocop doesn't like the way the old code had an addition operator/method within a safe navigation chain. I found that using `send` keeps Rubocop happy, but I wonder if there is a nicer way.

### CODE REVIEW

**Review by amichal — APPROVED** [review]
Looks good :)

File: app/models/unique_name.rb [hunk 1]
```diff
@@ -5,17 +5,23 @@
...
 
 class UniqueName < ApplicationRecord
-
   def self.update!
     Rails.logger.info 'Updating the unique names table'
 
     # Build double metaphone representations for all names in the database
-    names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.map{ |n| n&.downcase || ''} + GrdaWarehouse::Hud::Client.source.select('LastName').distinct.pluck('LastName')&.map{ |n| n&.downcase || ''}
-    names.uniq.each do |name|
+    existing_names = UniqueName.pluck(:name)
```
Comment (amichal) [inline app/models/unique_name.rb:12]: I wonder. Is the name property actually constrained to be unique in the database? Not that it matters here but curious about that

File: app/models/unique_name.rb [hunk 2]
```diff
@@ -5,17 +5,23 @@
...
+    all_names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.
```
Comment (amichal) [inline app/models/unique_name.rb]: It was in the code you inherited but I think `GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')` has identical behaviour as `GrdaWarehouse::Hud::Client.source.distinct.pluck('FirstName')` with less repetition

**Review by amichal — APPROVED** [review]

File: app/models/unique_name.rb (hunk 2 above)
Comment (kwiliarty) [reply app/models/unique_name.rb]: Cool. I did wonder about that.

**Comment by kwiliarty** [comment]
I've incorporated comments and some tweaks from our call. Thanks!

File: app/models/unique_name.rb [hunk 3]
```diff
@@ -5,17 +5,19 @@
...
-
   def self.update!
     Rails.logger.info 'Updating the unique names table'
 
     # Build double metaphone representations for all names in the database
-    names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.map{ |n| n&.downcase || ''} + GrdaWarehouse::Hud::Client.source.select('LastName').distinct.pluck('LastName')&.map{ |n| n&.downcase || ''}
-    names.uniq.each do |name|
+    existing_names = UniqueName.pluck(:name)
+    all_names = GrdaWarehouse::Hud::Client.source.distinct.pluck('FirstName').map { |n| n.to_s.downcase } +
+      GrdaWarehouse::Hud::Client.source.distinct.pluck('LastName').map { |n| n.to_s.downcase }
```
Comment (eanders) [inline app/models/unique_name.rb]: I would maybe write these as `GrdaWarehouse::Hud::Client.source.distinct.where.not(FirstName: [nil, '']).pluck('FirstName').map(&:downcase)`

Comment (kwiliarty) [reply app/models/unique_name.rb]: ✅ Thanks for the tip. I've made and tested the adjustment.

File: app/models/unique_name.rb (hunk 1 above)
Comment (eanders) [reply app/models/unique_name.rb:12]: Looks like not currently.

**Pull request merged**